marine/src/config/path_utils.rs

```diff
@@ -22,12 +22,15 @@ use thiserror::private::PathAsDisplay;
 use std::path::Path;
 use std::path::PathBuf;
 
-pub fn adjust_path(base: &Path, path: &Path) -> MarineResult<PathBuf> {
+pub fn as_relative_to_base(base: Option<&PathBuf>, path: &Path) -> MarineResult<PathBuf> {
```

Comment:
`&PathBuf` is basically the same as `Path`, is it not? Won't `Option<Path>` work here? I would expect it to work.

Reply:
`Option<Path>` does not work at all, unfortunately. Also i used Option<&PathBuf> because it is simpler to create it from `Option<PathBuf>`

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=be660d023f6e6835bfaab89390b60a0e